components/partials/swap/index.vue

```diff
@@ -0,0 +1,1574 @@
+        this.form.amount = new BigNumberInBase(form.amount).toFixed(
+          market.quantityDecimals,
+          BigNumberInBase.ROUND_DOWN
+        )
+      }
+    },
+
+    onTradingTypeToggle(selectedTradingType: TradeExecutionType) {
+      this.tradingType = selectedTradingType
+      this.setRateToLastTradePrice()
+    },
+
+    setRateToLastTradePrice() {
+      const { middlePrice } = this
+
+      if (!middlePrice) {
+        return
+      }
+
+      this.form.price = middlePrice.toString()
+    },
+
+    submitLimitOrder() {
+      const { orderType, market, price, amount } = this
+
+      if (!market) {
+        return
+      }
+
+      this.status.setLoading()
+
+      this.$accessor.spot
+        .submitLimitOrder({
+          price,
+          quantity: amount,
+          orderType
+        })
+        .then(() => {
+          this.$toast.success(this.$t('trade.swap.swap_success'))
+          this.$set(this, 'form', initialForm())
+        })
+        .catch(this.$onRejected)
+        .finally(() => {
+          this.status.setIdle()
+        })
+    },
+
+    submitMarketOrder() {
+      const { orderType, market, executionPrice, amount } = this
+
+      if (!market) {
+        return
+      }
+
+      this.status.setLoading()
+
+      this.$accessor.spot
+        .submitMarketOrder({
+          quantity: amount,
+          price: executionPrice,
+          orderType
+        })
+        .then(() => {
+          this.$toast.success(this.$t('trade.swap.swap_success'))
+          this.$set(this, 'form', initialForm())
+        })
+        .catch(this.$onRejected)
+        .finally(() => {
+          this.status.setIdle()
+        })
+    },
+
+    onSubmit() {
+      const {
+        hasErrors,
+        tradingTypeMarket,
+        priceHasHighDeviationWarning,
+        isUserWalletConnected
+      } = this
+
+      if (!isUserWalletConnected) {
+        return this.$toast.error(this.$t('please_connect_your_wallet'))
+      }
+
+      if (hasErrors) {
+        return this.$toast.error(this.$t('trade.error_in_form'))
+      }
+
+      if (tradingTypeMarket) {
+        return this.submitMarketOrder()
+      }
+
+      if (!priceHasHighDeviationWarning) {
+        return this.submitLimitOrder()
+      }
+
+      // If price has high deviation, we open a confirm modal
+      return this.$accessor.modal.openModal(Modal.OrderConfirm)
+    },
+
+    onSetAmount (amount: string) {
+      if (this.orderTypeBuy) {
+        this.buyAmount = amount
+        const amountInBase = new BigNumberInBase(amount)
+        const percent = amountInBase.div(this.quoteAvailableBalance).times(100).toNumber()
+        this.form.amount = this.getAmountByPercentage(percent)
+        this.form.toAmount = this.getAmountByPercentage(percent)
+      } else {
+        this.form.amount = amount
+      }
+    },
+
+    onSetToAmount(amount: string) {
+      this.form.toAmount = amount
+    },
+
+    onMaxInput(percent = 100) {
```

Comment:
as described in https://github.com/InjectiveLabs/injective-dex/pull/676/files#r845947143

lets do this instead:
```
onMaxInput() {
    const maxAmount = this.getAmountByPercentage(100)
}
```